appendices/appendix06-mimc-security.tex

```diff
@@ -0,0 +1,74 @@
+\end{itemize}
+
+Understanding the relationships between these various settings is required in order to use $\mimc{}$ to operate over prime fields of odd characteristic with a non-cubic permutation monomial.
+
+Overall, for $\mimc{}$ to be considered secure, it is important that no attack in the literature (that may provide a significant speedup compared to ``exhaustive key search'') can succesfully be mounted by a $\ppt$ adversary. Two main families of attacks exist: statistical attacks and algebraic attacks.
+
+\paragraph{Statistical attacks.}
+
+In the ``Security Analysis'' section~\cite[Section 4.2]{albrecht2016mimc}, the authors explain that since the cubic function is an Almost Perfect Nonlinear map (\apn)\footnote{A function $f(x) = x^{\exponent{}}$ over $\FFx{p^n}$ is said to be differentially $k$-uniform if $k$ is the maximum number of solutions $x \in \FFx{p^n}$ of $f(x + d) - f (x) = D$ where $d, D \in \FFx{p^n}$ and $d \neq 0$. If $f$ is a 2-uniform mapping, we say that $f$ is almost perfect nonlinear. See e.g.~\cite{DBLP:journals/tit/HellesethRS99} for more information.}, linear attacks pose no threat to $\mimc{}$.
+
+We observe that this claim aligns with~\cite[Theorem 2]{DBLP:journals/tit/HellesethRS99}. In fact if $\mimc{}$ is operated over $\FFx{2^n}$, it is easy to see that the degree of the cubic map can be expressed as $3 = 2^t + 1$, where $t = 1$, and where $n$ and $1$ are trivially coprime. This case is covered by~\cite[Theorem 2]{DBLP:journals/tit/HellesethRS99} which confirms that the cubic function $S(x) = x^3$ over $\FFx{2^n}$ is an APN/2-uniform mapping, as desired for differential and linear cryptanalysis resistance.
+
+Likewise, in~\cite[Section 5.1]{albrecht2016mimc} the authors claim that, provided that the cubic map is a permutation over the prime field of interest, $\mimc{}$ can be used to operate over prime fields of odd characteristic. In this case too, $S(x) = x^3$ is an APN, provided $p \neq 3$ (as reported by~\cite[Theorem 3, item 3]{DBLP:journals/tit/HellesethRS99}).
+
+In~\cite[Section 5.3]{albrecht2016mimc} the choice of the map degree is relaxed to be of the general form $2^t \pm 1$. Unfortunately, the authors showed that the case $\exponent{} = 2^t + 1$ is not as good as it initially seems in $\FFx{2^n}$, due to term cancellation in fields of characteristic 2 that renders the resulting polynomial sparse\footnote{Since the round function is a polynomial, the whole scheme can be seen as a polynomial which overall degree and ``sparsity'' depend on the underlying field characteristic, degree of the round function and number of rounds.}. More precisely the degree of the polynomial will be bounded by $3^r$, $r$ being the number of rounds, which does not constitute an improvement on the case of $\exponent{} = 3$. For this very reason, exponents of the form $2^t + 1, t > 1$, may not be of interest (sparse polynomial and more expensive arithmetic in the round function).
```

Comment:
`polynomial which overal degree` -> `polynomial with overall degree`?

```suggestion
In~\cite[Section 5.3]{albrecht2016mimc} the choice of the map degree is relaxed to be of the general form $2^t \pm 1$. Unfortunately, the authors showed that the case $\exponent{} = 2^t + 1$ is not as good as it initially seems in $\FFx{2^n}$, due to term cancellation in fields of characteristic 2 that renders the resulting polynomial sparse\footnote{Since the round function is a polynomial, the whole scheme can be seen as a polynomial with overall degree and ``sparsity'' that depends on the underlying field characteristic, degree of the round function and number of rounds.}. More precisely the degree of the polynomial will be bounded by $3^r$, $r$ being the number of rounds, which does not constitute an improvement on the case of $\exponent{} = 3$. For this very reason, exponents of the form $2^t + 1, t > 1$, may not be of interest (sparse polynomial and more expensive arithmetic in the round function).
```